zerver/actions/realm_settings.py

```diff
@@ -486,3 +487,67 @@ def do_send_realm_reactivation_email(realm: Realm, *, acting_user: Optional[User
         language=language,
         context=context,
     )
+
+
+def do_send_realm_deactivation_email(realm: Realm, acting_user: Optional[UserProfile]) -> None:
+    # if realm is not deactivated, but changed the domain
+    if not acting_user:
+        return
+
+    event_time = timezone_now()
+    RealmAuditLog.objects.create(
+        realm=realm,
+        acting_user=acting_user,
+        event_type=RealmAuditLog.REALM_DEACTIVATION_EMAIL_SENT,
+        event_time=event_time,
+        extra_data=orjson.dumps(
+            {
+                RealmAuditLog.ROLE_COUNT: realm_user_count_by_role(realm),
+            }
+        ).decode(),
+    )
+    owners = realm.get_human_owner_users()
+    admins = realm.get_human_admin_users(include_realm_owners=False)
+    admin = admins[0]
+    context = {
+        "realm_uri": realm.uri,
+        "realm_name": realm.name,
+        "event_date": event_time.date(),
+        "acting_user_name": acting_user.full_name,
+        "self_hosted": realm.PLAN_TYPE_SELF_HOSTED,
+        "admin_user_mail": admin.delivery_email,
+    }
+    language = realm.default_language
+
+    deactivating_owner = None
+    other_owners = []
+    for owner in owners:
+        if owner == acting_user:
+            deactivating_owner = owner
+        else:
+            other_owners.append(owner)
+
+    # Send separate emails to the deactivating owner and the rest of the owners
+    if deactivating_owner:
+        # Send email to deactivating owner
+        send_email(
+            "zerver/emails/realm_deactivated_self",
+            to_user_ids=[deactivating_owner.id],
```

Comment:
we can directly use `to_emails` instead of `to_user_ids`.
Reason: If you check the implementation of `send_email`, internally it prepares `to_emails` from `to_user_ids`. We should save that repetitive work.

Reply:
Hmm, sounds good.